### ISSUE TITLE
block-scoped-var has false positive

### ISSUE THREAD

**Issue by mysticatea** [mysticatea/eslint-plugin#7]
This is a repro code:

```js
var a = 0

module.exports = a
```

If a reference exists at the end of a file, it's warned as "not defined".
This happens only with semicolon-less style.

The cause is [here](https://github.com/mysticatea/eslint-plugin/blob/b7e2beaef7db622b19816b75a4fa51d17f9341be/lib/rules/block-scoped-var.js#L167).
